src/plot/series/canvas-wrapper.js

```diff
@@ -21,47 +21,150 @@
+ * @param {CanvasContext} ctx - the context where the drawing will take place
+ * @param {Number} height - height of the canvas
+ * @param {Number} width - width of the canvas
+ * @param {Array} layers - the layer objects to render
+ * @param {Number} drawIteration - width of the canvas
+ */
+function drawLayers(ctx, height, width, layers, drawIteration) {
+  ctx.clearRect(0, 0, width, height);
+  layers.forEach(layer => {
+    const {interpolator, newProps, animation} = layer;
+    // return an empty object if dont need to be animating
+    const interpolatedProps = animation ?
+      (interpolator ? interpolator(drawIteration / MAX_DRAWS) : interpolator) :
+      () => ({});
+    layer.renderLayer({
+      ...newProps,
+      ...interpolatedProps
+    }, ctx);
+  });
+}
+
+/**
+ * Build an array of layer of objects the contain the method for drawing each series
+ * as well as an interpolar (specifically a d3-interpolate interpolator)
+ * @param {Object} newChildren the new children to be rendered.
+ * @param {Object} oldChildren the old children to be rendered.
+ * @returns {Array} Object for rendering
+ */
+function buildLayers(newChildren, oldChildren) {
+  return newChildren.map((child, index) => {
+    const oldProps = oldChildren[index] ? oldChildren[index].props : {};
+    const newProps = child.props;
+
+    const oldAnimatedProps = extractAnimatedPropValues({
+      ...oldProps,
+      animatedProps: ANIMATED_SERIES_PROPS
+    });
+    const newAnimatedProps = newProps ? extractAnimatedPropValues({
+      ...newProps,
+      animatedProps: ANIMATED_SERIES_PROPS
+    }) : null;
+    const interpolator = interpolate(oldAnimatedProps, newAnimatedProps);
+
+    return {
+      renderLayer: child.type.renderLayer,
+      newProps: child.props,
+      animation: child.props.animation,
+      interpolator
+    };
+  });
+}
 class CanvasWrapper extends Component {
   componentDidMount() {
-    this.drawChildren(this.props, this.refs.canvas.getContext('2d'));
+    this.drawChildren(this.props, null, this.refs.canvas.getContext('2d'));
   }
 
   componentWillUpdate(nextProps) {
-    this.drawChildren(nextProps, this.refs.canvas.getContext('2d'));
+    this.drawChildren(nextProps, this.props, this.refs.canvas.getContext('2d'));
   }
 
-  drawChildren(props, ctx) {
+  drawChildren(newProps, oldProps, ctx) {
     const {
       children,
       innerHeight,
-      innerWidth
-    } = props;
+      innerWidth,
+      marginBottom,
+      marginLeft,
+      marginRight,
+      marginTop
+    } = newProps;
     if (!ctx) {
       return;
     }
-    ctx.clearRect(0, 0, innerWidth, innerHeight);
-    children.forEach(layer => layer.type.renderLayer(layer.props, ctx));
+
+    const childrenShouldAnimate = checkChildrenForAnimation(children);
+
+    const height = innerHeight + marginTop + marginBottom;
+    const width = innerWidth + marginLeft + marginRight;
+    const layers = buildLayers(newProps.children, oldProps ? oldProps.children : []);
```

Comment:
I'm not sure i get what you are suggesting? oldProps is necessary for interpolation against the previous state?